client/components/jetpack/jetpack-presales-chat-widget/index.tsx

```diff
@@ -50,9 +55,30 @@ export const ZendeskJetpackChat: React.VFC< { keyType: KeyType } > = ( { keyType
 			: ! isLoggedIn && isEnglishLocale && isCorrectContext && isStaffed;
 	}, [ isStaffed, isLoggedIn, keyType ] );
 
-	if ( error ) {
-		return null;
-	}
+	useEffect( () => {
+		if ( ! zendeskChatKey || isLoadingAuth ) {
+			return;
+		}
+
+		const result = loadScript(
+			'https://static.zdassets.com/ekr/snippet.js?key=' + encodeURIComponent( zendeskChatKey ),
+			undefined,
+			{ id: 'ze-snippet' }
+		);
+
+		//pass authentication key to Zendesk
+		Promise.resolve( result ).then( () => {
+			if ( typeof window !== 'undefined' ) {
+				window.zE = window.zE || [];
+
+				if ( zendeskdJwt && typeof window.zE === 'function' ) {
+					window.zE( 'messenger', 'loginUser', function ( callback: ( jwt: string ) => void ) {
+						callback( zendeskdJwt );
+					} );
+				}
+			}
```

Comment:
Non-blocking nit-picks:

- Inverting this conditional will help readability by reducing nesting.
- Extra inline documentation can't hurt :-)
- I think it's safe for us to get rid of the safe assignment of `window.zE` to an array, if we're checking first that it's a function before continuing.

```suggestion
			// We need the user's authentication token to log them into chat
			if ( ! zendeskdJwt ) {
				return;
			}
			
			// Chat can't exist if we're not in a browser window
			if ( typeof window === 'undefined' ) {
				return;
			}
			
			// The `zE` function exposes the required action to authenticate the user
			if ( typeof window.zE !== 'function' ) {
				return;
			}

			window.zE(
				'messenger',
				'loginUser',
				function ( callback: ( jwt: string ) => void ) {
					callback( zendeskdJwt );
				}
			);
```

Reply:
I'm just gonna wholesale steal that :)